### PR TITLE
Pass stateless_alu from command line

### PR DESCRIPTION
**Pull request by XiangyuG**

### CODE REVIEW

File: chipc/direct_solver.py [hunk 1]
```diff
@@ -37,10 +39,11 @@ def main(argv):
     args = parser.parse_args(argv[1:])
 
     sketch_name = args.program_file.split('/')[-1].split('.')[0] + \
-        "_" + args.alu_file.split('/')[-1].split('.')[0] + \
+        "_" + args.stateful_alu_file.split('/')[-1].split('.')[0] + \
         "_" + str(args.num_pipeline_stages) + \
```
Comment (anirudhSK) [inline chipc/direct_solver.py]: Can we also add the stateless ALU file name to the sketch_name string?

File: chipc/direct_solver.py [hunk 2]
```diff
@@ -37,10 +39,11 @@ def main(argv):
     args = parser.parse_args(argv[1:])
 
     sketch_name = args.program_file.split('/')[-1].split('.')[0] + \
-        "_" + args.alu_file.split('/')[-1].split('.')[0] + \
+        "_" + args.stateful_alu_file.split('/')[-1].split('.')[0] + \
```
Comment (anirudhSK) [inline chipc/direct_solver.py:42]: Can we also add the stateless_alu_file to the sketch_name string?

File: chipc/iterative_solver.py [hunk 3]
```diff
@@ -121,10 +123,11 @@ def main(argv):
     state_group_info = get_info_of_state_groups(program_content)
 
     sketch_name = args.program_file.split('/')[-1].split('.')[0] + \
-        "_" + args.alu_file.split('/')[-1].split('.')[0] + \
+        "_" + args.stateful_alu_file.split('/')[-1].split('.')[0] + \
```
Comment (anirudhSK) [inline chipc/iterative_solver.py:126]: Same as the comment above.

File: chipc/sketch_generator.py [hunk 4]
```diff
@@ -76,8 +78,10 @@ def add_assert(self, assert_predicate):
 
     # Generate Sketch code for a simple stateless alu (+,-,*,/)
     def generate_stateless_alu(self, alu_name, potential_operands):
+        # Grab the stateless alu file name by using
+        # self.stateless_alu_file_[self.stateless_alu_file_.rfind('/')+1:]
         stateless_alu_template = self.jinja2_env_.get_template(
-            "stateless_alu.j2")
+            self.stateless_alu_file_[self.stateless_alu_file_.rfind('/')+1:])
```
Comment (anirudhSK) [inline chipc/sketch_generator.py:85]: Why can't we just pass in the stateless ALU file? The rfind seems brittle. What if the stateless ALU file is not in the same directory as the repository?

File: tests/test_chipmunk.py [hunk 5]
```diff
@@ -134,15 +149,19 @@ def test_simple_2_2_raw_cex_mode(self):
             iterative_solver.main([
                 "iterative_solver",
                 path.join(SPEC_DIR, "simple.sk"),
-                path.join(ALU_DIR, "raw.stateful_alu"), "2", "2"]),
+                path.join(ALU_DIR, "raw.stateful_alu"),
```
Comment (anirudhSK) [inline tests/test_chipmunk.py:151]: Rename ALU_DIR to STATEFUL_ALU_DIR.

File: chipc/sketch_generator.py (hunk 4 above)
Comment (taegyunkim) [reply chipc/sketch_generator.py:85]: Use [os.path.split()](https://docs.python.org/3/library/os.path.html#os.path.split) instead.

File: chipc/sketch_generator.py [hunk 6]
```diff
@@ -76,8 +78,10 @@ def add_assert(self, assert_predicate):
 
     # Generate Sketch code for a simple stateless alu (+,-,*,/)
     def generate_stateless_alu(self, alu_name, potential_operands):
+        # Grab the stateless alu file name by using
```
Comment (taegyunkim) [inline chipc/sketch_generator.py]: Remove redundant code.

Comment (taegyunkim) [inline chipc/sketch_generator.py:82]: Remove redundant comment. This and following line.

**Review by taegyunkim — CHANGES_REQUESTED** [review]

**Pull request closed without merging**